### ISSUE TITLE
Accessing selectionEnd on an input element that cannot have a selection

### ISSUE THREAD

**Issue by mockdeep** [honeybadger-io/honeybadger-js#34]
Ugh, now that we're one level deeper we're now seeing this error:

```
Accessing selectionEnd on an input element that cannot have a selection.
```

It looks like part of the serialization process ends up trying to iterate across a DOM node at times and comes across a property that doesn't want to be touched. Came across this SO question: http://stackoverflow.com/questions/11547672/how-to-stringify-event-object

**Comment by joshuap** [comment]
hmm, do you have any idea how the event object is getting added to the honeybadger payload? i.e. is it being added to the context or something? I don't think we should normally be including stuff like that by default.

**Comment by joshuap** [comment]
Also, is the error when converting the payload to JSON, or does it happen in the `_santize` method? If the latter, maybe I could just try/catch errors when accessing keys in that method and return an object like `{ "error": "Accessing selectionEnd on an input element that cannot have a selection." }` instead.

**Comment by mockdeep** [comment]
Oh, hmm, I guess I thought the error being serialized might have a DOM node in there somewhere. We don't add it to the context in our application. It's turning up within the `_sanitize` method, so I think your idea of adding the try/catch would do the trick.

**Comment by joshuap** [comment]
Do you think it could have anything to do with [accessing `navigator`](https://github.com/honeybadger-io/honeybadger-js/blob/master/src/notice.coffee#L51)?

**Comment by joshuap** [comment]
Although I guess in that statement we are excluding objects entirely.

**Comment by mockdeep** [comment]
As far as I can tell it seems to be happening within `_sanitize` [here in the compiled source](https://github.com/honeybadger-io/honeybadger-js/blob/master/honeybadger.js#L153). Not super helpful, since we don't have source maps, but our trace looks like this:

```
TypeError: Accessing selectionEnd on an input element that cannot have a selection.
    at t._sanitize (https://chalkschools.com/assets/honeybadger_config-5c406bce047f50f0c686273298e2875b.js:1:2182)
    at t._sanitize (https://chalkschools.com/assets/honeybadger_config-5c406bce047f50f0c686273298e2875b.js:1:2340)
```

You can `wget` that file (while it's cached) and page over to column `2182` to see where it is coming from. We discover more js weirdness every day. This one has mostly been on various Android versions after 4, though we've also seen it on Chrome 11.

**Comment by joshuap** [comment]
How about 06f15125b1ef41cbdfd981436f23f74afb207268?

**Comment by mockdeep** [comment]
That looks like it'll do the trick.

**Comment by joshuap** [comment]
I've released 0.3.1.
